python/ray/util/sgd/v2/migration/sgd-migration-guide.md

```diff
@@ -0,0 +1,224 @@
+# Ray SGD: v1 to v2 migration guide
+
+In Ray 1.7, we are rolling out a new and more streamlined version of Ray SGD. Ray SGD v2 focuses on usability and composability- it has a much simpler API, has support for more deep learning backends, integrates better with other libraries in the Ray ecosystem, and will continue to be actively developed with more features. 
+
+This guide will cover to easily migrate existing code from Ray SGD v1 to Ray SGD v2. If you are new to Ray SGD as a whole, you should get started with Ray SGD v2 directly: https://docs.ray.io/en/master/raysgd/v2/raysgd.html. 
+
+For a full list of features that Ray SGD v2 provides, please check out the user guide: https://docs.ray.io/en/master/raysgd/v2/user_guide.html. 
+
+If there are any issues or anything missing with this guide or any feedback on Ray SGD v2 overall, please file a Github issue on the Ray repo: https://github.com/ray-project/ray/issues!
+
+## What are the API differences?
+There are 2 primary API differences between Ray SGD v1 and v2.
+
+1. There is a single `Trainer` interface for all backends (torch, tensorflow, horovod), and the backend is simply specified via an argument: `Trainer(backend="torch")`, `Trainer(backend="horovod")`, etc. Any features that we add to Ray SGD will be supported for all backends, and there won't be any API divergence like there was with a separate `TorchTrainer` and `TFTrainer`.
+2. The `TrainingOperator` and creator functions are replaced by a generic user-defined training function. You no longer have to make your training logic fit into a restrictive interface. In Ray SGD v2, you simply have to provide a training function that describes the full logic for your training execution and this will be distributed by Ray SGD. 
```

Comment:
```suggestion
2. The `TrainingOperator` and creator functions are replaced by a more natural user-defined training function. You no longer have to make your training logic fit into a restrictive interface. In Ray SGD v2, you simply have to provide a training function that describes the full logic for your training execution and this will be distributed by Ray SGD. 
```

Reply:
FIxed!